src/services/formatting/formatting.ts

```diff
@@ -626,6 +627,10 @@ namespace ts.formatting {
 
                 childContextNode = node;
 
+                if (isFirstListItem && parent.kind === SyntaxKind.ArrayLiteralExpression && inheritedIndentation === Constants.Unknown) {

```

Comment:
why do we need to specialize it to array, should not the firstListItem be enough indicator?